src/plugins/initiatives/initiative.js

```diff
@@ -1,9 +1,26 @@
 // @flow
 
-import {type NodeAddressT} from "../../core/graph";
+import {type NodeAddressT, NodeAddress} from "../../core/graph";
+import {initiativeNodeType} from "./declaration";
 
 export type URL = string;
 
+// Composite ID, used as input for NodeAddressT.
+export opaque type InitiativeId = string[];
+
+// Enforce that each ID has at least a sub-type and 1..n components.
+export function createId(
```

Comment:
Per [this review](https://github.com/sourcecred/sourcecred/pull/1648#pullrequestreview-355519335), I'd like to see a method that constructs an Id from an InitiativesDirectory and a InitiativeMapEntry -- maybe it could go into this PR? (Could use this function as a helper.)

Reply:
I'll add it to the other PR :]